nf_core/modules/module_test.py

```diff
@@ -0,0 +1,145 @@
+        self.pytest_args = pytest_args
+
+    def run(self):
+        """Run test steps"""
+        if not self.no_prompts:
+            log.info(
+                "[yellow]Press enter to use default values [cyan bold](shown in brackets) [yellow]or type your own responses"
+            )
+        self._check_inputs()
+        self._set_profile()
+        self._check_profile()
+        self._run_pytests()
+
+    def _check_inputs(self):
+        """Do more complex checks about supplied flags."""
+
+        # Get the tool name if not specified
+        if self.module_name is None:
+            if self.no_prompts:
+                raise UserWarning(
+                    "Tool name not provided and prompts deactivated. Please provide the tool name as TOOL/SUBTOOL or TOOL."
+                )
+            modules_repo = ModulesRepo()
+            modules_repo.get_modules_file_tree()
+            self.module_name = questionary.autocomplete(
+                "Tool name:",
+                choices=modules_repo.modules_avail_module_names,
+                style=nf_core.utils.nfcore_question_style,
+            ).ask()
+        module_dir = Path("modules") / self.module_name
+
+        # First, sanity check that the module directory exists
+        if not module_dir.is_dir():
+            raise UserWarning(
+                f"Cannot find directory '{module_dir}'. Should be TOOL/SUBTOOL or TOOL. Are you running the tests inside the nf-core/modules main directory?"
+            )
+
+    def _set_profile(self):
+        """Set $PROFILE env variable.
+        The config expects $PROFILE and Nextflow fails if it's not set.
+        """
+        if os.environ.get("PROFILE") is None:
+            os.environ["PROFILE"] = ""
+            if self.no_prompts:
+                log.info(
+                    "Setting environment variable '$PROFILE' to an empty string as not set.\n"
+                    "Tests will run with Docker by default. "
+                    "To use Singularity set 'export PROFILE=singularity' in your shell before running this command."
+                )
+            else:
+                question = {
+                    "type": "list",
+                    "name": "profile",
+                    "message": "Choose software profile",
+                    "choices": ["Docker", "Singularity", "Conda"],
+                }
+                answer = questionary.unsafe_prompt([question], style=nf_core.utils.nfcore_question_style)
+                profile = answer["profile"].lower()
+                os.environ["PROFILE"] = profile
+                log.info(f"Setting environment variable '$PROFILE' to '{profile}'")
+
+    def _check_profile(self):
+        """Check if profile is available"""
+        profile = os.environ.get("PROFILE")
+        # Make sure the profile read from the environment is a valid Nextflow profile.
+        valid_nextflow_profiles = ["docker", "singularity", "podman", "shifter", "charliecloud", "conda"]
```

Comment:
Support for `$PROFILE` in nf-core/modules != support for container tech in Nextflow:

https://github.com/nf-core/modules/blob/801240a9714082de913e8a72f4bb434018a50e8b/tests/config/nextflow.config#L14-L22

```suggestion
        valid_nextflow_profiles = ["docker", "singularity", "conda"]
```